examples/ex1_2d_full_control.py

```diff
@@ -0,0 +1,183 @@
+    # The field being analysed as used by your input script 
+    field_key = "temperature"
+    # Analysis sensor type 
+    # (thermocouples for temperature, disp_sensors for displacement, or strain_gauges for strain)
+    sensor_type = "thermocouples"
+    # Sensor arrangement (x_sensors, y_sensors, z_sensors)
+    sensors = (3,2,1)
+    # Number of spatial dimensions being used 
+    dims = 2
+    # Whether the sensors should include basic errors or not
+    errors = False
+    # Whether the model should be a multi-classifier rather than binary, so it can 
+    # distinguish between perturbations to different classes of invalid parameters 
+    # (geometry, BCs, material properties) rather than just valid and invalid results
+    multi = True 
+    # Whether the unlabelled data should be deleted 
+    delete_datasets = False
+    # Whether the model should be saved as a .pkl file and what it should be called
+    save = False
+    model_name = "ex1_2d_model"
+
+    # Setup MOOSE aspects of Icarus
+    moose_setup = MooseSetup(input_file_path, n_tasks=n_tasks, n_threads=n_threads)
+    moose_runner, moose_modifier = moose_setup.setup_moose_runner()
+
+    # Accesses the available parameters 
+    found_vars = moose_modifier.get_vars()
+    if len(parameters) == 0:
+        # If none have been manually specified, allow user to select which parameters are perturbed, 
+        # what values they should take, and how many validation values to use for each
+        num_validation_values, parameters = UserInterface().accept_parameters(found_vars)
+    else:
+        # Ensure parameters dictionary and num_validation_values list are valid 
+        if len(num_validation_values) != len(parameters) or 0 in num_validation_values:
+            print("Invalidation number of validation values specified")
+            sys.exit()
+        for param_name, param_data in parameters.items():
+            if param_name not in found_vars.keys():
+                print(f"Parameter {param_name} not found in input file. Exiting.")
+                sys.exit() 
+            if param_data[0].lower() not in ["geom", "bc", "mat_prop"]:
+                print(f"Invalid parameter class. Exiting.")
+                sys.exit()
+            if len(param_data[1]) < 1:
+                print("Insufficient parameter values. Exiting.")
+                sys.exit()
+    
+    # Initialise dataset generator and generate perturbed, validation and ground truths datasets
+    dataset_generator = DatasetGenerator(moose_runner, moose_modifier, parameters, output_file_path, num_para_runs)
+
+    # Sets up the paths to where the outputted unlabelled datasets will be saved
+    perturbed_path, perturbed_vals = Path(str(output_file_path+"perturbed_datasets/")), None
+    validation_path, validation_vals = Path(str(output_file_path+"validation_datasets/")), None
+    
+    paths = {perturbed_path: perturbed_vals, validation_path: validation_vals}
+
+    # Generates the unlabelled datasets
```

Comment:
```suggestion
    # Generate the unlabelled datasets
```
Try to keep the comments consistently in the imperative mood (to remember how that works, think of it like a command, a to-do list item, or finishing the sentence "When you run this code, it will...").

Reply:
Quite a few examples of this, I didn't leave a comment on each one but worth a pass through to correct them.